File: WORK/ClassWork_260916/green_tree[45].py
import sys

number = int(sys.argv[1])
Matrix = [[0 for x in range(number)] for y in range(number)] 

def generate_green_tree(number):
	stop = 0
	for i in reversed(range(0,number)):
		for j in range(0,number):
			if (j >= stop and j < number - stop):
				Matrix[i][j] = 1
				print ("i: " + str(i) + " j: " + str(j) + " stop: " + str(stop) + " filled")
			else:
				Matrix[i][j] = 0
				print ("i: " + str(i) + " j: " + str(j) + " stop: " + str(stop) + "not filled")
		if (number-((stop+1)*2) >= 1):
			stop += 1


def generate_html_table(inputarray):
	output = "<table class='green-tree square-table'>\n"
	for j in range(0,number):
		output += "\t<tr>\n"
		for i in range(0,number):
			output += generate_cell(inputarray[j][i]);

		output += "\t</tr>\n"
	output += "</table>\n"
	return output


def generate_cell(isFilled):
	output = "\t\t<td class='"
	if isFilled:
		output += "filled-zero-cell"
	else:
		output += "filled-one-cell"
	output +=  "' width='"
	output += str(int(round(200/number,0)))
	output += "' heights='"
	output += str(int(round(200/number,0)))
	output += "' />\n"
	return output


generate_green_tree(number)
print generate_html_table(Matrix)

File: WORK/ClassWork_260916/green-tree[45].js
var matrix = [];
var number = +process.argv[2];

function generate_green_tree(number){


	stop = 0;
	for (var i=number-1;i>=0;i--){
		matrix[i] = [];
		for (var j = 0; j < number; j++){
			if (j >= stop && j < number - stop){
				matrix[i][j] = 1;
			}
			else{
				matrix[i][j] = 0;
			}
		}
		if (number-((stop+1)*2) >= 1){
			stop += 1;
		}
	}

}


function generate_html_table(inputarray){
	output = "<table class='green-tree square-table'>\n";
	for (var j = 0; j < number; j++){
		console.log(inputarray[4]);
		output += "\t<tr>\n";
		for (var i = 0; i < number; i++){
			console.log(inputarray[j][i]);
			output += generate_cell(inputarray[j][i]);
		}

		output += "\t</tr>\n";
	}
	output += "</table>\n";
	return output;
}


function generate_cell(isFilled){
	output = "\t\t<td class='";
	if (!isFilled){
		output += "filled-zero-cell";
	}
	else{
		output += "filled-one-cell";
	}
	output +=  "' width='";
	output += Math.round(200/number,0);
	output += "' heights='";
	output += Math.round(200/number,0);
	output += "' />\n";
	return output;
}


generate_green_tree(number)
console.log(matrix)
console.log(generate_html_table(matrix))
console.log(matrix)